### DISCUSSION TITLE
Alternative way to select uppercase arguments - No Caps Lock

### DISCUSSION THREAD

**Discussion by zakissimo** [NeogitOrg/neogit#710] [Q&A] [answered]
In my system, I have `caps lock` remapped to `ctrl` . Which means I can't trigger uppercase arguments in neogit. For some reason holding `shift` + letter isn't working. Am I missing something ? Is there a way for me to trigger uppercase flags/arguments without using `caps lock` ?

**Comment by CKolkey** [comment]
Shift+letter definitely should work. I have the same caps-lock-is-ctrl setup, and no issues. 

That said, flags and arguments have a prefix, either a minus or equal sign. You're including that?

**Reply by ten3roberts** [reply]
I am using a layered keyboard with colemak , so I don't have a caps-lock at all.

If I do `P` to get to the push popup, I can do

`-` followed by `F` to toggle the force option.

Does that work for you?

**Comment by zakissimo** [comment]
Minus followed by shift + letter isn't working for me. Only when I reactivate CapsLock, and use it, I'm able to trigger the flags.

If you say it should work, I must have some uppercase letter keybind. I'll investigate. Thanks for the quick replies.

**Reply by zakissimo** [reply, marked as answer]
Okay, turns out, it works for me too ... I have `vim.opt.timeoutlen = 350` in my settings and I guess every time I tried to input `-F` it took me more than 350 ms to go from pressing `-` to pressing `shift + f` ...
Sorry for the inconvenience ...

**Reply by CKolkey** [reply]
No worries, glad we could sort it out :)

**Reply by IndianBoy42** [reply]
I had the same problem. what I ended up doing is 
```lua
      local group = vim.api.nvim_create_augroup ("NeogitUserAucmds", {})
      vim.api.nvim_create_autocmd("FileType", {
        group = "NeogitUserAucmds",
        pattern = "NeogitPopup",
        callback = function()
          require("which-key").register({
            ["-"] = "Options",
            ["="] = "Arguments",
          }, { buffer = 0 })
        end,
      })
```

Registering those prefixes with which-key, which means I get the which-key popup but more importantly that means it waits for my next keypress for as long as I want. It would be nice if the keymappings by neogit had `desc = ` in the `keymap.set` opts so the which-key menu was prettier. Mini.clue would also work, or hydra.nvim maybe, or some handrolled mapping based on `getchar` similar to what which-key/mini.clue does

**Reply by CKolkey** [reply]
Would that really be helpful? Theres already a pop-up that shows you the key and a description of what it does... feels kinda like putting a hat on a hat, you know?

**Reply by IndianBoy42** [reply]
yeah I'm not doing it for the popup but for the "wait as long as the user wants" for the next key press. Just that the popup shows a bunch of ugly lua for now
![image](https://github.com/NeogitOrg/neogit/assets/5981889/c12d354b-e6d9-43b9-81c8-52b18496c9e4)

**Reply by CKolkey** [reply]
How's this work instead? https://github.com/NeogitOrg/neogit/pull/747
